internal/nonkube/bundle_extract.go

```diff
@@ -0,0 +1,69 @@
+package nonkube
+
+import (
+	"bytes"
+	_ "embed"
+	"fmt"
+	"os"
+	"path"
+	"text/template"
+)
+
+type SelfExtractingBundle struct {
+	SiteName   string
+	OutputPath string
+}
+
+func (s *SelfExtractingBundle) InstallFile() string {
+	return path.Join(s.OutputPath, fmt.Sprintf("skupper-install-%s.sh", s.SiteName))
+}
+
+func (s *SelfExtractingBundle) Generate(tarBall *Tarball) error {
+	var data = new(bytes.Buffer)
+	var err error
+
+	write := func(buf interface{}) error {
+		var size, written int
+		switch buf.(type) {
```

Comment:
assigning the result of type assertions can prevent bugs doing type assertions in the case statements below.

ex: https://go.dev/play/p/TSTaFNXFQHR

Reply:
Fixed. Thanks for the feedback.